From 6ab2cc667468284ef176b72859823e2e946e9ee6 Mon Sep 17 00:00:00 2001
From: Shuicheng Lin <shuicheng.lin@intel.com>
Date: Fri, 7 Feb 2020 15:21:23 +0800
Subject: [PATCH] Fix queue_lock compile issue

queue_lock was "spinlock_t *" type in "struct request_queue". But
with current 4.18.0-80 kernel, it is "spinlock_t".

Signed-off-by: Shuicheng Lin <shuicheng.lin@intel.com>
---
 drbd/drbd_main.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drbd/drbd_main.c b/drbd/drbd_main.c
index 4dd0bf1..8022c9a 100644
--- a/drbd/drbd_main.c
+++ b/drbd/drbd_main.c
@@ -2910,7 +2910,7 @@ enum drbd_ret_code drbd_create_device(struct drbd_config_context *adm_ctx, unsig
 
 	drbd_init_set_defaults(device);
 
-	q = blk_alloc_queue(GFP_KERNEL);
+	q = blk_alloc_queue_node(GFP_KERNEL, NUMA_NO_NODE);
 	if (!q)
 		goto out_no_q;
 	device->rq_queue = q;
@@ -2944,7 +2944,6 @@ enum drbd_ret_code drbd_create_device(struct drbd_config_context *adm_ctx, unsig
 #ifdef COMPAT_HAVE_BLK_QUEUE_MERGE_BVEC
 	blk_queue_merge_bvec(q, drbd_merge_bvec);
 #endif
-	q->queue_lock = &resource->req_lock;
 #ifdef blk_queue_plugged
 		/* plugging on a queue, that actually has no requests! */
 	q->unplug_fn = drbd_unplug_fn;
-- 
2.7.4

